### PR TITLE
necessary changes for Hopper-ising Routemaster

### PR DESCRIPTION
**Pull request by humzashah**
status: work in progress. need to fix the branches for which circleCI builds.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,56 +1,184 @@
+version: 2
+
 defaults: &defaults
   working_directory: /routemaster
   docker:
-    - image: deliveroo/circleci:0.1.0
+    - image: deliveroo/circleci:0.1.8
```
Comment (bencord0) [inline .circleci/config.yml]: The latest version is `0.1.15`.
https://github.com/deliveroo/circleci/blob/master/VERSION

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,56 +1,184 @@
...
 
-version: 2
+build_steps: &build_steps
+  steps:
+    - setup_remote_docker:
+        reusable: true
+        version: 17.05.0-ce
```
Comment (bencord0) [inline .circleci/config.yml]: `17.11.0-ce` is available
https://circleci.com/docs/2.0/building-docker-images/#docker-version

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,56 +1,184 @@
...
-  run_test_suite:
-    <<: *defaults
+    - checkout
 
-    # we want to run tests multiple times to minimise the risk of flaky tests in
-    # this (critical) application
-    parallelism: 5
+    - run:
+        name: Clean up env
+        command: ci down
```
Comment (bencord0) [inline .circleci/config.yml]: Do you need this step?

File: .circleci/config.yml [hunk 4]
```diff
@@ -1,56 +1,184 @@
...
+
+    - run:
+        name: Build Docker image
+        command: |
```
Comment (bencord0) [inline .circleci/config.yml]: I see you are using the `ci` helper for tests.
Could you also use `ci build && ci tag` here?

(`ci tag` is a new feature in the `0.1.15` image of `deliveroo/circleci`)

File: Dockerfile [hunk 5]
```diff
@@ -1,5 +1,9 @@
+FROM deliveroo/hopper-runner:1.0.0 as hopper-runner
```
Comment (bencord0) [inline Dockerfile]: The latest version is `1.2.0`.
https://github.com/deliveroo/hopper-runner/blob/master/VERSION

**Comment by humzashah** [comment]
Closing this in favour of https://github.com/deliveroo/routemaster/pull/111 . Struggled getting tests to pass here so started a new branch.

**Pull request closed without merging**